src/main/java/com/amazon/dlic/auth/http/saml/HTTPSamlAuthenticator.java

```diff
@@ -184,7 +184,9 @@ public Optional<SecurityResponse> reRequestAuthentication(final SecurityRequest
             if (API_AUTHTOKEN_SUFFIX.equals(suffix)) {
                 // Verficiation of SAML ASC endpoint only works with RestRequests
                 if (!(request instanceof OpenSearchRequest)) {
-                    throw new SecurityRequestChannelUnsupported();
+                    throw new SecurityRequestChannelUnsupported(
+                        API_AUTHTOKEN_SUFFIX + " not supported for request of type " + request.getClass().getName()
```

Comment:
we should add/modify-existing tests to check for this messaged

Reply:
There are a couple of already filed issues to example our SAML tests in greater detail:

1. https://github.com/opensearch-project/security/issues/2989
2. https://github.com/opensearch-project/security/issues/3212

This PR has discussion on uncovering the permissions issues encountered after the OpenSAML upgrade:

- https://github.com/opensearch-project/security/pull/2987

Reply:
@cwperks @DarshitChanpura What would you like to do about this comment thread?

Reply:
I think this can be marked as resolved. There are tests that hit this endpoint, but the tests can be enhanced in this repo as the linked issues mention.

Reply:
resolve it :D